Easydict/Swift/Service/Gemini/GeminiService.swift

```diff
@@ -123,11 +122,116 @@ public final class GeminiService: LLMStreamService {
         Defaults[.geminiAPIKey] ?? ""
     }
 
+    override var availableModels: [String] {
+        Defaults[.geminiValidModels]
+    }
+
+    override var model: String {
+        get {
+            Defaults[.geminiModel]
+        }
+        set {
+            // easydict://writeKeyValue?EZGeminiModelKey=gemini-1.5-flash
+            Defaults[.geminiModel] = newValue
+        }
+    }
+
     // MARK: Private
 
     // Set Gemini safety level to BLOCK_NONE
     private let harassmentBlockNone = SafetySetting(harmCategory: .harassment, threshold: .blockNone)
     private let hateSpeechBlockNone = SafetySetting(harmCategory: .hateSpeech, threshold: .blockNone)
     private let sexuallyExplicitBlockNone = SafetySetting(harmCategory: .sexuallyExplicit, threshold: .blockNone)
     private let dangerousContentBlockNone = SafetySetting(harmCategory: .dangerousContent, threshold: .blockNone)
+
+    private func promptContent(
+        queryType: EZQueryTextType,
+        text: String,
+        from sourceLanguage: Language,
+        to targetLanguage: Language
+    )
+        -> [ModelContent] {
+        var prompts = [[String: String]]()
+
+        switch queryType {
+        case .dictionary:
+            prompts = dictMessages(word: text, sourceLanguage: sourceLanguage, targetLanguage: targetLanguage)
+        case .sentence:
+            prompts = sentenceMessages(sentence: text, from: sourceLanguage, to: targetLanguage)
+        case .translation:
+            fallthrough
+        default:
+            prompts = translationMessages(text: text, from: sourceLanguage, to: targetLanguage)
+        }
+
+        var chats: [ModelContent] = []
+        for prompt in prompts {
+            if let roleRaw = prompt["role"],
+               let parts = prompt["content"] {
+                let role = getCorrectParts(from: roleRaw)
+                let chat = ModelContent(role: role, parts: parts)
+                chats.append(chat)
+            }
+        }
+        guard !chats.isEmpty else {
+            return chats
+        }
+        // removing first element in [ModelContent] since it's system instruction
+        chats.removeFirst()
```

Comment:
This code is not good. If we need to add other LLM services later, there will be similar bad code.

We can optimize the code in `Prompt.swift` by extracting `systemMessages`.

Reply:
I added an argument to the Messages functions in `Prompts` to tell if a LLM services needs to add systemPrompts

Reply:
ok, seems good.